pandas/conftest.py

```diff
@@ -881,3 +881,72 @@ def index_or_series(request):
     See GH#29725
     """
     return request.param
+
+
+_indexes = {
+    "bool-index": tm.makeBoolIndex(10, name="a"),
+    "int-index": tm.makeIntIndex(10, name="a"),
+    "float-index": tm.makeFloatIndex(10, name="a"),
+    "date-index": tm.makeDateIndex(10, name="a"),
+    "localized-date-index": tm.makeDateIndex(10, name="a", tz="US/Eastern"),
+    "period-index": tm.makePeriodIndex(10, name="a"),
+    "string-index": tm.makeStringIndex(10, name="a"),
+    "unicode-index": tm.makeUnicodeIndex(10, name="a"),
+}
+
+
+@pytest.fixture(params=_indexes.keys())
+def index(request):
+    """
+    Fixture for tests on indexes
+
+    copy to avoid mutation, e.g. setting .name
+    """
+    return _indexes[request.param].copy()
+
+
+_arr = np.random.randn(10)
+
+
+@pytest.fixture
+def int_series():
+    """ Fixture for Series with random numbers and integer index """
+    index = _indexes["int-index"].copy()
+    return pd.Series(_arr, index=index, name=index.name)
+
+
+_series = {
+    f"series-with-{i_id}": pd.Series(_arr, index=i, name=i.name)
+    for i_id, i in _indexes.items()
+}
+
+_arr_int = np.random.choice(10, size=10, replace=False)
+
+
+def _create_narrow_series(data, data_dtype):
+    """ Helper for the _narrow_series dict """
+    index = _indexes["int-index"].copy()
+    return pd.Series(data.astype(data_dtype), index=index, name="a")
```

Comment:
this needs to conform with pandas/tests/indexes/conftest.py (likely easier to pull that up to top-level)